src/iuse_actor.cpp

```diff
@@ -112,9 +112,7 @@ long iuse_transform::use(player *p, item *it, bool t, const tripoint &pos ) cons
         // "created" right now and give the content the current time as birthday
         it->convert( container_id );
         it->unset_curammo();
-        it->charges = -1;
-        it->contents.push_back(item(target_id, calendar::turn));
-        target = &it->contents.back();
+        target = &it->emplace_in( target_id, calendar::turn, -1 );
```

Comment:
Uhm, doesn't that set the charges of `target` to -1, not the charges of `it` (as it was done previously)?
Further more: doesn't a value of -1 as `qty` parameter of `item::item` mean to set the *default* charges? It's even the default value of that parameter anyway, so why specify it at all?

----

In case you didn't know: this code is for opening a sealed jar of food. The jar used to have charges (was a tool), converting it back into a plain jar (`it->convert(container_id)`) kept the charges, but plain jars don't need charges (they are generic items).

Instead of setting charges to -1, it should set charges to 0 (as this is the new default). Or don't change them at all because "sealed jars of foo" are now generic items as well, and therefor don't have any charges that need to be reset.